File: Scadex.Model/Dtos/User/Queries/UserDetailDto.cs
using Scadex.Core.Model;

namespace Scadex.Model.Dtos.User.Queries;

public class UserDetailDto : IDto
{
    public Guid Id { get; set; }
    public string? Email { get; set; }
    public Guid CompanyId { get; set; }
    public string CompanyName { get; set; } = null!;
    public string FullName { get; set; } = null!;
    public string? PhoneNumber { get; set; }
    public string? CreatedBy { get; set; }
    public string? UpdatedBy { get; set; }
    public DateTime? CreateDateUtc { get; set; }
    public DateTime? UpdateDateUtc { get; set; }
    public bool IsActive { get; set; }
}

File: Scadex.WebUI/src/models/user/queries/userDetailDto.ts
/** Ayna: Scadex.Model/Dtos/User/Queries/UserDetailDto.cs */
export interface UserDetailDto {
  id: string;
  /** Giriş adı — ad soyad tekil değildir, kullanıcıyı ayırt eden bu. */
  userName: string | null;
  email: string | null;
  companyId: string;
  companyName: string;
  fullName: string;
  phoneNumber: string | null;
  createdBy: string | null;
  updatedBy: string | null;
  /** `Z` soneki YOK (`datetime2`) — gösterilecekse `toUtcDate` ile. */
  createDateUtc: string | null;
  updateDateUtc: string | null;
  isActive: boolean;
}
